sdk/storage/Azure.Storage.Common/src/Shared/Errors.Clients.cs

```diff
@@ -47,6 +47,12 @@ public static InvalidOperationException StreamMustBeAtPosition0()
         public static InvalidOperationException TokenCredentialsRequireHttps()
             => new InvalidOperationException("Use of token credentials requires HTTPS");
 
+        public static ArgumentException SasCredentialRequiresUriWithoutSas<TUriBuilder>(Uri uri)
```

Comment:
@jkotas, how does it affect AOT scenarios? Would it be better to pass the type as a method parameter (object or Type)?

Reply:
Generic arguments are better than passing Type around. It is not just AOT, it is also a for IL trimming. Passing System.Type around is a potential problem for IL trimming and may sometime need annotations to work well with it.

Reply:
In this specific case, it should not really matter since the typeof usage is trivial and the method is small. https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/ThrowHelper.cs in the CoreLib uses both patterns.

Reply:
Great. so this stays I guess.